statistics/handle/handle.go

```diff
@@ -426,10 +426,10 @@ func (h *Handle) MergePartitionStats2GlobalStats(sc *stmtctx.StatementContext, i
 			}
 			globalStats.Hg[i].NDV = globalStatsNDV
 		} else {
-			// For the index stats, we get the final NDV by accumulating the NDV of each bucket in the index histogram.
+			// For the index stats, we get the final NDV by accumulating the NDV of each partition histogram.
 			globalStatsNDV := int64(0)
-			for _, bucket := range globalStats.Hg[i].Buckets {
-				globalStatsNDV += bucket.NDV
+			for _, ph := range allHg[i] {
+				globalStatsNDV += ph.NDV
```

Comment:
Why change here? The bucket NDV will change in the `MergePartitionHist2GlobalHist`. It seems that we should use the bucket ndv in the merged histogram to calculate.

Reply:
`bucket.NDV` is not maintained when `stats-version` is `version1`, so the result will be zero if we accumulate `bucket.NDV`.
Compared with accumulating `bucket.NDV`, accumulating `partition.Hist.NDV` is more general and can work on both `stats-version`.

Reply:
We can't use `partition.Hist.NDV` to calculate the `global NDV` because data in different partitions will overlap.